app_feup/lib/view/Pages/general_page_view.dart

```diff
@@ -102,12 +84,20 @@ abstract class GeneralPageViewState extends State<StatefulWidget> {
       backgroundColor: Theme.of(context).backgroundColor,
       titleSpacing: 0.0,
       title: ButtonTheme(
-          minWidth: 0,
-          padding: EdgeInsets.only(left: 0),
-          materialTapTargetSize: MaterialTapTargetSize.shrinkWrap,
-          shape: RoundedRectangleBorder(),
-          child: FlatButton(
-            onPressed: () => Navigator.pushNamed(context, '/' + Constants.navPersonalArea),
+        minWidth: 0,
+        padding: EdgeInsets.only(left: 0),
+        materialTapTargetSize: MaterialTapTargetSize.shrinkWrap,
+        shape: RoundedRectangleBorder(),
+        child: FlatButton(
+            onPressed: () {
+              final currentRouteName = ModalRoute.of(context).settings.name;
+              if(currentRouteName != Constants.navPersonalArea){
+                Navigator.popUntil(
+                  context,
+                  ModalRoute.withName(Constants.navPersonalArea)
+                );
```

Comment:
popUntil doesn't seem to work in this case whenever there's nothing else in the stack. Isn't there anything else better?

Reply:
It might honestly be better for the UX tbh....

Reply:
Done